### PR TITLE
[INJIMOB-1862] deprecate verifyCredentials and introduce verify method

### PR DESCRIPTION
**Pull request by KiruthikaJeyashankar**
Major change: 
- introduce new method for verify and deprecate older method
verifyCredentials method has been deprecated because it is not extensible for future use cases of supporting different VC format's verification. To support this usecase new method has been introduced -

              verify(credential: String, credentialFormat: CredentialFormat): Boolean

      The introduction of the new method `verify` will be used further to identify the verification logic that needs to be applied over the `credential` based on the input `credentialFormat`, providing support for other credential formats (as per CredentialFormat constant exposed)

Other changes:
- Design change : introduce factory pattern for vc verification based on format

### CODE REVIEW

File: vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt [hunk 1]
```diff
@@ -1,175 +1,28 @@
...
-import java.security.cert.CertificateException
-import java.security.spec.MGF1ParameterSpec
-import java.security.spec.PSSParameterSpec
-import java.security.spec.X509EncodedKeySpec
-import java.util.Objects
+import io.mosip.vercred.vcverifier.constants.CredentialFormat
+import io.mosip.vercred.vcverifier.credentialverifier.CredentialVerifierFactory
 
 
 class CredentialsVerifier {
```
Comment (KiruthikaJeyashankar) [inline vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt:8]: Need inputs on of if the class name `CredentialsVerifier` needs modification, If change is required will be creating a new class which will eventually have the new method - verify

File: vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt [hunk 2]
```diff
@@ -1,175 +1,28 @@
...
-                    val pubKeySpec = X509EncodedKeySpec(pubKeyBytes)
-                    val keyFactory = KeyFactory.getInstance("RSA")
-                    keyFactory.generatePublic(pubKeySpec)
-                } else null
-            }
-        } catch (e: Exception) {
-            Log.e(tag, "Error Generating public key object", e)
-            null
-        }
+    fun verify(credential: String, credentialFormat: CredentialFormat): Boolean {
```
Comment (balachandarg-tw) [inline vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt:25]: If we are getting the whole VC with format and credential, why don't we pass that whole VC to the library as single param and extracting out the format in library itself ?

File: vc-verifier/kotlin/vcverifier/src/test/java/io/mosip/vercred/vcverifier/credentialverifier/types/LdpVcCredentialVerifierTest.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @After
+    fun after() {
+        clearAllMocks()
+    }
+
+    @Test(timeout = 10000)
+    fun `should return true for valid credential`() {
+        val validVC =
```
Comment (balachandarg-tw) [inline vc-verifier/kotlin/vcverifier/src/test/java/io/mosip/vercred/vcverifier/credentialverifier/types/LdpVcCredentialVerifierTest.kt:31]: Is it fine to push valid VC ?

File: vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt (hunk 1 above)
Comment (swatigoel) [reply vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt:8]: Leave this classname as is

**Review by swatigoel — APPROVED** [review]

File: vc-verifier/kotlin/vcverifier/src/test/java/io/mosip/vercred/vcverifier/credentialverifier/types/LdpVcCredentialVerifierTest.kt (hunk 3 above)
Comment (swatigoel) [reply vc-verifier/kotlin/vcverifier/src/test/java/io/mosip/vercred/vcverifier/credentialverifier/types/LdpVcCredentialVerifierTest.kt:31]: yes, have already discussed it with Anusha and Vishwa.
These is qa env so valid test VC

File: vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt (hunk 2 above)
Comment (swatigoel) [reply vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/CredentialsVerifier.kt:25]: After draft13 OpenId4VCI, VC response doesn't include format

**Pull request merged**